core/src/main/scala/org/bykn/bosatsu/TypedExpr.scala

```diff
@@ -20,6 +20,7 @@ sealed abstract class TypedExpr[T] {
    * for each expression
    *
    */
+  def updatedTag(t: T): TypedExpr[T]
```

Comment:
This is always implemented using `copy`. There is surely a more direct way to implement this but I couldn't figure out how

Reply:
I don't know of one. But Nit: can we implement this with a pattern match rather than a method to be consistent with the other code?

This can be a PR that can be merged immediately.